kep/640-error-pods-rescheduling/README.md

```diff
@@ -0,0 +1,166 @@
+# Error pods rescheduling
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+- [Design Details](#design-details)
+  - [Rescheduler-controller](#rescheduler-controller)
+    - [Scheduled nodes list and current rescheduling times definition](#scheduled-nodes-list-and-current-rescheduling-times-definition)
+  - [Podrescheduling](#podrescheduling)
+  - [A concrete use case](#a-concrete-use-case)
+  - [Notes](#notes)
+- [Known limitations](#known-limitations)
+- [Test Plan](#test-plan)
+- [Graduation Criteria](#graduation-criteria)
+  - [Alpha](#alpha)
+  - [Beta](#beta)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+## Summary
+
+This proposal outlines a solution for rescheduling pods that have encountered errors, such as pods being OOM-killed or experiencing errors related to NUMA or devices.
```

Comment:
We're working on a [similar design](https://docs.google.com/document/d/1-wJhiNy84w7tzFdo9HqwTu5DrVSuXFLGTUv8FBiRAAc/edit?usp=sharing) spawned from DRA beta graduation conversation and it's very hard to implement this approach without kubelet changes, and even with that we need to take into account races with existing controllers.

Because this proposal is against scheduler-plugins, this suggests you are trying to avoid core kubernetes changes. Is my understanding right?

Reply:
Thanks for your review! Yes, we don't need to change any core kubernetes code, and just implement a scheduler-plugin.